food-for-thought/app/user.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { View, Text, TouchableOpacity } from 'react-native';
+import { Card, Icon } from '@rneui/themed';
+import { router } from 'expo-router';
+import Header from "@/components/Header";     
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { jwtDecode } from 'jwt-decode';
+import { styles } from '../styles/app-styles';
+import axios from 'axios';
+import Constants from 'expo-constants';
+
+const UserProfile = () => {
+  const [username, setUsername] = useState('');
+  const [isGuest, setIsGuest] = useState(true);
+  const [userNotes, setUserNotes] = useState(0);
+  const [userFavourites, setUserFavourites] = useState(0);
+  const [userPreferences, setUserPreferences] = useState(0);
+
+  const loadUser = useCallback(async () => {
+    const token = await AsyncStorage.getItem('token');
+    if (token) {
+      try {
+        const decodedToken: any = jwtDecode(token);
+        setUsername(decodedToken.username);
+        setIsGuest(false);
+      } catch (error) {
+        console.error("Invalid token");
+      }
+    }
+  }, []);
+
+  useEffect(() => {
+    loadUser();
+    handleUserDetails();
+  }, [loadUser]);
+
+  const handleLogout = async () => {
+    await AsyncStorage.removeItem('token');
+    setUsername('');
+    setIsGuest(true);
+    router.push('/');
+  };
+
+  const HOST_IP = Constants.expoConfig?.extra?.HOST_IP;
+
+  const handleUserDetails = useCallback(async () => {
+    try {
+      if (!isGuest) {
+        const response = await axios.get(`http://${HOST_IP}:4000/user/getUser/${username}`);
+        setUserFavourites(response.data?.favourites?.length ?? 0);
+        setUserNotes(response.data?.notes?.length ?? 0);
+        setUserPreferences(response.data?.preferences?.length ?? 0);
+      }
+    } catch (error: any) {
+      console.error(error);
+    }
+  }, [userNotes, userFavourites, userPreferences]);
+
+  return (
+    <View style={styles.container}>
+      <Header />
```

Comment:
Is there a way to keep the header aligned to the top, to be consistent with the other pages?
<img width="200" alt="image" src="https://github.com/user-attachments/assets/101fe74e-4e99-4d9a-bdbd-bee8619513c4"> <img width="200" alt="image" src="https://github.com/user-attachments/assets/51656ae5-d840-42c6-933b-1f31d6ec5746">

Reply:
This does show up as aligned on my phone but my phone is smaller than average. Will attempt again on the web interface!